packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -66,7 +67,12 @@ class DaemonCommand extends FlutterCommand {
   }
 
   dynamic _handleError(dynamic error, StackTrace stackTrace) {
-    printError('Error from flutter daemon: $error', stackTrace: stackTrace);
+    if (error is ToolExit) {
+      printError(error.message);
+      exit(error.exitCode);
```

Comment:
Yeah, in my local edits, I removed this error handler - not sure why it was there...